mara_db/shell.py

```diff
@@ -329,33 +375,54 @@ def __(db: dbs.SqlcmdSQLServerDB, header: bool = None, footer: bool = None, deli
 
 
 @copy_to_stdout_command.register(dbs.OracleDB)
-def __(db: dbs.OracleDB, header: bool = None, footer: bool = None, delimiter_char: str = None, csv_format: bool = None):
+def __(db: dbs.OracleDB, header: bool = None, footer: bool = None, delimiter_char: str = None, csv_format: bool = None,
+       pipe_format: Format = None):
     assert all(v is None for v in [header, footer, delimiter_char, csv_format]), "unimplemented parameter for OracleDB"
+    if pipe_format and not isinstance(pipe_format, CsvFormat):
+        raise ValueError(f'Unsupported pipe_format for OracleDB: {pipe_format}')
     return "(echo 'set markup csv on\nset feedback off\nset heading off' && cat)" \
            + " \\\n  | " + query_command(db)
 
 
 @copy_to_stdout_command.register(dbs.SQLiteDB)
-def __(db: dbs.SQLiteDB, header: bool = None, footer: bool = None, delimiter_char: str = None, csv_format: bool = None):
+def __(db: dbs.SQLiteDB, header: bool = None, footer: bool = None, delimiter_char: str = None, csv_format: bool = None,
+       pipe_format: Format = None):
+    assert all(v is None for v in [footer, csv_format]), "unimplemented parameter for SQLiteDB"
+
+    if pipe_format:
+        if isinstance(pipe_format, CsvFormat):
+            if header is None:
+                header = pipe_format.header
+            if delimiter_char is None:
+                delimiter_char = pipe_format.delimiter_char
+        else:
+            raise ValueError(f'Unsupported pipe_format for SQLiteDB: {pipe_format}')
+
     if header is None:
         header = False
 
     if delimiter_char is None:
         delimiter_char = '\t'
 
-    assert all(v is None for v in [footer, csv_format]), "unimplemented parameter for SQLiteDB"
     header_argument = '-noheader' if not header else ''
     return query_command(db) + " " + header_argument + f" -separator '{delimiter_char}' -quote"
 
 
 @copy_to_stdout_command.register(dbs.SnowflakeDB)
-def __(db: dbs.SnowflakeDB, header: bool = None, footer: bool = None, delimiter_char: str = None, csv_format: bool = None):
+def __(db: dbs.SnowflakeDB, header: bool = None, footer: bool = None, delimiter_char: str = None, csv_format: bool = None,
+       pipe_format: Format = None):
     assert footer is None, "unimplemented parameter for SnowflakeDB"
 
-    if csv_format is None:
-        csv_format = True
+    if pipe_format:
+        if isinstance(pipe_format, CsvFormat):
+            if header is None:
+                header = pipe_format.header
+            if delimiter_char is None:
+                delimiter_char = pipe_format.delimiter_char
+        else:
+            raise ValueError(f'Unsupported pipe_format for SnowflakeDB: {pipe_format}')       
 
-    if not csv_format:
+    if csv_format == False:
```

Comment:
Nice find! Although I would remove the "not set" from the error, as it seems the code should be right?